compiler/ghc/default.nix

```diff
@@ -68,12 +68,14 @@ let
 
   # TODO check if this posible fix for segfaults works or not.
   targetLibffi =
-    if stdenv.targetPlatform.isMusl
```

Comment:
couldn't we do 
```
let targetLibffi = targetPackages.libffi or libffi # <some comment explaining the rational for targetPackages.libffi not being there>
 in if stdenv.targetPlatform.isMusl then targetLibffi.override... else targetLibffi;
```?

Reply:
There is no targetPackages.libffi or targetPackages.gmp for pkgsCross.musl64, they are just libffi and gmp:
```
$ nix-env -f '<nixpkgs>' -qaP -A pkgsCross.musl64.libffi
pkgsCross.musl64.libffi  libffi-3.3-x86_64-unknown-linux-musl
$ nix-env -f '<nixpkgs>' -qaP -A pkgsCross.musl64.buildPackages.libffi
pkgsCross.musl64.buildPackages.libffi  libffi-3.3
$ nix-env -f '<nixpkgs>' -qaP -A pkgsCross.musl64.targetPackages.libffi
error: attribute 'libffi' in selection path 'pkgsCross.musl64.targetPackages.libffi' not found
```
targetPackages in pkgsCross (musl64, raspberryPi, etc) always empty

Reply:
So the current (new) logic is:
```
targetLibffi =
  if stdenv.targetPlatform.isMusl && (targetPackages.libffi or null) != null
  then targetPackages.libffi.overrideAttrs (old: { dontDisableStatic = true; })
  else if targetPlatform != hostPlatform
  then (targetPackages.libffi or libffi)
  else libffi;
```
isn't this equivalent to:
```
targetLibffi = 
  # on native platforms targetPlatform.{libffi, gmp} do not exist; thus fall back
  # to the non-targetPlatform version in those cases.
  let targetLibffi = targetPackages.libffi or libffi; in
  # we need to set `dontDisableStatic` for musl for libffi to work.
  if stdenv.targetPlatform.isMusl
  then targetLibffi.override (old: { dontDisableStatic = true; })
  else targetLibffi;
```

Reply:
Yes, this looks much better, thanks.